src/cascadia/UnitTests_TerminalCore/SelectionTest.cpp

```diff
@@ -94,8 +94,123 @@ namespace TerminalCoreUnitTests
+
+            // Test DoubleClickSelection(COORD)
+            // Behavior: clamp coord to viewport.
+            //           Then, do double click selection.
+            auto ValidateDoubleClickSelection = [&](SHORT scrollback, SMALL_RECT expected) {
+                Terminal term;
+                DummyRenderTarget emptyRT;
+                term.Create({ 10, 10 }, scrollback, emptyRT);
+
+                term.DoubleClickSelection(maxCoord);
+                ValidateSelection(term, expected);
+            };
+
+            // Test TripleClickSelection(COORD)
+            // Behavior: clamp coord to viewport.
+            //           Then, do triple click selection.
+            auto ValidateTripleClickSelection = [&](SHORT scrollback, SMALL_RECT expected) {
+                Terminal term;
+                DummyRenderTarget emptyRT;
+                term.Create({ 10, 10 }, scrollback, emptyRT);
+
+                term.TripleClickSelection(maxCoord);
+                ValidateSelection(term, expected);
+            };
+
+            // Test with no scrollback
+            ValidateSingleClickSelection(0, { 9, 9, 9, 9 });
+            ValidateDoubleClickSelection(0, { 0, 9, 9, 9 });
+            ValidateTripleClickSelection(0, { 0, 9, 9, 9 });
+
+            // Test with max scrollback
+            const SHORT expected_row = SHRT_MAX - 1;
+            ValidateSingleClickSelection(SHRT_MAX, { 9, expected_row, 9, expected_row });
+            ValidateDoubleClickSelection(SHRT_MAX, { 0, expected_row, 9, expected_row });
+            ValidateTripleClickSelection(SHRT_MAX, { 0, expected_row, 9, expected_row });
+        }
+
+        TEST_METHOD(SelectFromOutofBounds)
+        {
+            /*  NOTE:
+                ensuring that the selection anchors are clamped to be valid permits us to make the following assumption:
+                    - All selection expansion functions will operate as if they were performed at the boundary
+            */
+
+            Terminal term;
+            DummyRenderTarget emptyRT;
+            term.Create({ 10, 10 }, 0, emptyRT);
+
+            auto viewport = term.GetViewport();
+            const SHORT leftBoundary = viewport.Left();
+            const SHORT rightBoundary = viewport.RightInclusive();
+            const SHORT topBoundary = viewport.Top();
+            const SHORT bottomBoundary = viewport.BottomInclusive();
+
+            auto ValidateSelection = [&](SMALL_RECT expected) {
+                auto selectionRects = term.GetSelectionRects();
+
+                // Validate selection area
+                VERIFY_ARE_EQUAL(selectionRects.size(), static_cast<size_t>(1));
+                auto selection = viewport.ConvertToOrigin(selectionRects[0]).ToInclusive();
+
+                VERIFY_ARE_EQUAL(selection, expected);
+            };
+
+            // Case 1: Simulate click past right (x,y) = (20,5)
+            // should clamp to right boundary
+            term.SetSelectionAnchor({ 20, 5 });

```

Comment:
Probably a few log subcomments here too so we can see which specific subcase failed.